### PR TITLE
[REF] mail: bring dataChannels model closer to the rtc implementation

### PR DESCRIPTION
**Pull request by ThanhDodeurOdoo**
This commit puts the rtcDataChannels field on RtcPeerConnection which
is closer to the webRtc implementation.

### CODE REVIEW

File: addons/mail/static/src/models/rtc.js [hunk 1]
```diff
@@ -170,7 +170,6 @@ registerModel({
                 rtcSession.update({
                     connectionRecoveryTimeout: clear(),
                     isCurrentUserInitiatorOfConnectionOffer: clear(),
-                    rtcDataChannel: clear(),
```
Comment (ThanhDodeurOdoo) [inline addons/mail/static/src/models/rtc.js:173]: because now cleared at #151

File: addons/mail/static/src/models/rtc_data_channel.js [hunk 2]
```diff
@@ -16,8 +16,12 @@ registerModel({
             required: true,
             readonly: true,
         }),
-        rtcSession: one('RtcSession', {
-            inverse: 'rtcDataChannel',
+        label: attr({
+            required: true,
+            readonly: true,
+        }),
```
Comment (alexkuhn) [inline addons/mail/static/src/models/rtc_data_channel.js:22]: Missing docstring. This is especially important for non-relational fields.

File: addons/mail/static/src/models/rtc.js [hunk 3]
```diff
@@ -474,7 +473,8 @@ registerModel({
             rtcSession.update({ rtcPeerConnection: insertAndReplace({ peerConnection }) });
             this.messaging.models['RtcDataChannel'].insert({
                 dataChannel,
-                rtcSession: replace(rtcSession),
+                label: dataChannel.label,
+                rtcPeerConnection: replace(rtcSession.rtcPeerConnection),
```
Comment (alexkuhn) [inline addons/mail/static/src/models/rtc.js:477]: At a glance, `label` should be computed field and just alias for `RtcDataChannel/dataChannel/label`...

File: addons/mail/static/src/models/rtc.js [hunk 4]
```diff
@@ -644,8 +644,11 @@ registerModel({
                     if (!rtcSession) {
                         continue;
                     }
-                    const rtcDataChannel = rtcSession.rtcDataChannel;
-                    if (!rtcDataChannel || rtcDataChannel.dataChannel.readyState !== 'open') {
+                    const rtcDataChannel = this.messaging.models['RtcDataChannel'].insert({
+                        label: "notifications",
+                        rtcPeerConnection: replace(rtcSession.rtcPeerConnection),
+                    });
```
Comment (alexkuhn) [inline addons/mail/static/src/models/rtc.js:650]: ...But then why the `label` is forged here, and don't just use `RtcDataChannel/dataChannel/label`?

**Pull request closed without merging**